dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/RenovateUtil.java

```diff
@@ -0,0 +1,103 @@
+package com.salesforce.dockerfileimageupdate.utils;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.salesforce.dockerfileimageupdate.CommandLine;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.bouncycastle.util.io.pem.PemReader;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.GitHubBuilder;
+import org.kohsuke.github.HttpException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.Security;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Instant;
+import java.util.Date;
+
+public class RenovateUtil {
+    private static final Logger log = LoggerFactory.getLogger(RenovateUtil.class);
+
+    private static String appId;
+    private static String privateKeyPath;
+    private static String jwt;
+    private static Instant jwtExpiry;
+    private static GitHub gitHub;
+
+    public RenovateUtil(final Namespace ns){
+        this.appId = ns.get(Constants.RENOVATE_GITHUB_APP_ID);
+        this.privateKeyPath = ns.get(Constants.RENOVATE_GITHUB_APP_KEY);
+        jwt = null;
+        jwtExpiry = null;
+        gitHub = null;
+        try {
+            generateJWT(this.appId, this.privateKeyPath);
+        } catch (GeneralSecurityException | IOException exception) {
+            log.warn("Could not initialise JWT due to exception: {}", exception.getMessage());
+        }
+        try {
+            gitHub = new GitHubBuilder()
+                    .withEndpoint(CommandLine.gitApiUrl(ns))
+                    .withJwtToken(jwt)
+                    .build();
+        } catch (IOException exception) {
+            log.warn("Could not initialise github due to exception: {}", exception.getMessage());
+        }
+    }
+
+    protected boolean isRenovateEnabledOnRepository(String fullRepoName){
```

Comment:
let's make this more generic - to check for any app existance

Reply:
Done.